.github/workflows/ci.yml

```
@@ -57,40 +60,83 @@ jobs:
           ~/.cargo/registry/cache/
           ~/.cargo/git/db/
           target/
-        key: ${{ runner.os }}-cargo-${{ steps.args.outputs.args }}-${{ hashFiles('**/Cargo.lock') }}
+        key: ${{ runner.os }}-${{ runner.arch }}-cargo-${{ steps.args.outputs.args }}-${{ hashFiles('**/Cargo.lock') }}
     - run: cargo ${{ matrix.args }}
 
   container:
-    runs-on: ubuntu-24.04
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - amd64
+          - arm64
+    runs-on: ${{ (matrix.arch == 'arm64' && 'ubuntu-24.04-arm') || 'ubuntu-24.04' }}
     steps:
     - uses: actions/checkout@v4
       with:
         submodules: true
+
+    - run: |
+        echo "FACT_TAG=$(make tag)-${{ matrix.arch }}" >> "$GITHUB_ENV"
+
     - run: make image
+
     - name: Login to quay.io/stackrox-io
       uses: docker/login-action@v3
       with:
         registry: quay.io
         username: ${{ secrets.QUAY_STACKROX_IO_RW_USERNAME }}
         password: ${{ secrets.QUAY_STACKROX_IO_RW_PASSWORD }}
+
     - run: docker push "$(make image-name)"
 
-  tag:
+  vars:
     runs-on: ubuntu-24.04
     outputs:
-      tag: ${{ steps.tag.outputs.tag }}
+      tag: ${{ steps.vars.outputs.tag }}
+      image-name: ${{ steps.vars.outputs.image-name }}
     steps:
     - uses: actions/checkout@v4
       with:
         submodules: true
-    - id: tag
-      run: echo "tag=$(make tag)" >> "$GITHUB_OUTPUT"
 
-  integration-tests:
+    - id: vars
+      run: |
+        {
+          echo "tag=$(make tag)"
+          echo "image-name=$(make image-name)"
+        } >> "$GITHUB_OUTPUT"
```

[nit] Same number of lines, but this is a bit easier to read IMO.

```suggestion
        cat << EOF >> "$GITHUB_OUTPUT"
        tag="$(make tag)"
        image-name="$(make image-name)"
        EOF
```